docs/proposals/rollout/rollout.md

```diff
@@ -0,0 +1,531 @@
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type RolloutPolicy struct {
+    // Checknum defines the number of checks to run for A/B Testing and Blue/Green
+    // Note: Kurator determines whether blue-green or A/B related processing is required based on 
+    // the presence or absence of content in the Checknum field. 
+    // So can't configure Iterations and CanaryStrategy at the same time.
+    // +optional
+    CheckNum int `json:"checknum,omitempty"`
+
+    // The TrafficRouting defines the configuration of the gateway, traffic distribution rules, and so on.
+    TrafficRouting *TrafficRoutingConfig `json:"trafficRouting"`
+
+    // TrafficAnalysis defines the validation process of a release
+    TrafficAnalysis *TrafficAnalysis `json:"trafficAnalysis,omitempty"`
+
+    // ProgressDeadlineSeconds represents the maximum time in seconds for a
+    // canary deployment to make progress before it is considered to be failed.
+    // Defaults to 600s.
+    // +optional
+    ProgressDeadlineSeconds *int32 `json:"progressDeadlineSeconds,omitempty"`
+
+    // SkipAnalysis promotes the canary without analyzing it
+    // +optional
+    SkipTrafficAnalysis bool `json:"skipTrafficAnalysis,omitempty"`
+
+    // Restore resources to initial state when deleting canary resources.
+    // Use of the revertOnDeletion property should be enabled
+    // when you no longer plan to rely on Kurator for deployment management.
+    // +optional
+    RevertOnDeletion bool `json:"revertOnDeletion,omitempty"`
+
+    // Suspend, if set to true will suspend the Canary, disabling any canary runs
+    // regardless of any changes to its target, services, etc. Note that if the
+    // Canary is suspended during an analysis, its paused until the Canary is uninterrupted.
+    // +optional
+    Suspend bool `json:"suspend,omitempty"`
+}
+```
+
+WorkloadReference contains enough information to let you locate the typed referenced object in the same namespace. The two types of Kind now supported are `Deployment` and `DaemonSet`.
+
+```console
+type WorkloadReference struct {
+    // API version of the referent
+    // +optional
+    APIVersion string `json:"apiVersion,omitempty"`
+
+    // Kind of the referent.
+    // Support Deployment and DaemonSet.
+    Kind string `json:"kind"`
+
+    // Namespace of the referent
+    Namespace string `json:"namespace"`
+
+    // Name of the referent
+    Name string `json:"name"`
+}
+```
+
+Kurator will create a VirtualService resource based on the configuration in `VirtualServiceConfig` to distribute traffic.
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type TrafficRoutingConfig struct {
+    // Timeout of the HTTP or gRPC request.
+    // Timeout in upstream response time.
+    // +optional
+    Timeout string `json:"timeout,omitempty"`
```

Comment:
Can you give more comments what should we set? 
what is a valid timeout？`5s`, or `5seconds`?

Reply:
`5s` is the valid timmeout. I will add an example.